### DISCUSSION TITLE
await session.BanGroupMemberAsync(context.GroupId, context.UserId, new TimeSpan(2000))这个是指定用户禁言吗？测试好像不起作用呢？

### DISCUSSION THREAD

**Discussion by 867525168** [OrgEleCho/EleCho.GoCqHttpSdk#28] [Q&A] [answered]
`await session.BanGroupMemberAsync(context.GroupId, context.UserId, new TimeSpan(2000))`
这个是指定用户禁言吗？测试好像不起作用呢？

**Comment by SlimeNull** [comment]
可用的, 机器人需要是管理员或群主才行. 如果没有起作用, 你可以看看 go-cqhttp 日志中是否有报错

**Comment by SlimeNull** [comment]
另外, 时间太短似乎是不能禁言的

**Comment by SlimeNull** [comment, marked as answer]
我注意到你使用的是 new TimeSpan, 它的间隔太小了, 甚至不到1s, 你可以使用 `TimeSpan.FromMinutes(10)` 来创建十分钟的时间间隔.
